netrd/dynamics/lotka_vottera.py

```diff
@@ -0,0 +1,135 @@
+
+        inter (np.ndarray): $N \times N$ array of interaction weights between
+                            nodes. If not specified, default to a zero-diagonal
+                            matrix whose [i, j] entry is $frac{sign(j - i)}{N - 1}$.
+
+        dt (float or np.ndarray): Sizes of time steps when simulating the
+                                  continuous-time dynamics.
+
+        stochastic (bool): Whether to simulate the stochastic or deterministic
+                           dynamics.
+
+        pertb (np.ndarray): Length-$N$ 1D array of perturbation magnitude of
+                            nodes' growth. If not specified, default to 0.01
+                            for all nodes.
+
+        Returns
+        -------
+        TS (np.ndarray): $N \times L$ array of $L$ observations on $N$ nodes.
+
+        Notes
+        -----
+        1. The deterministic dynamics is simulated through the forth-order
+           Runge-Kutta method, and the stochastic one is simulated through
+           mutliplicative noise with the Euler-Maruyama method.
+
+        2. The ground-truth network, time steps and the time series can be
+           found in results['ground-truth'], reuslts['time_steps'] and
+           results['time_series'] respectively.
+
+        """
+        N = G.number_of_nodes()
+        adjmat = nx.to_numpy_array(G)
+
+        # Initialize the model's parameters if not specified
+        if gr is None:
+            gr = np.ones(N, dtype=float)
+        if cap is None:
+            cap = np.ones(N, dtype=float)
+        if inter is None:
+            wei = 1 / (N - 1)
+            full = np.full((N, N), wei, dtype=float)
+            inter = np.zeros((N, N), dtype=float)
+            inter += np.triu(full) - np.tril(full)
+
+        if stochastic and pertb is None:
+            pertb = 1e-2 * np.ones(N, dtype=float)
+
+        # Randomly initialize an initial condition if not speciefied
+        TS = np.zeros((N, L), dtype=float)
+        if init is None:
+            init = unifrom(low=0, high=cap)
+        TS[:, 0] = init
+
+        # Define the function of dynamics
+        mat = np.where(adjmat == 1, inter, 0.0) + np.diag(-np.ones(N))
+        mat /= cap[:, np.newaxis]
+
+        def dyn(t, state):
+            return state * (gr + np.dot(mat, state))
+
+        # Simulate the time series
+        if isinstance(dt, float):
+            dt = dt * np.ones(L-1)
+
+        # Deterministic dynamics
+        if not stochastic:
+            integrator = ode(dyn).set_integrator('dropi5')
+            integrator.set_initial_value(init, 0.0)
+            for t in range(L-1):
+                if integrator.succesful():
+                    TS[:, t+1] = integrator.integrate(integrator.t + dt[t])
+                else:
+                    message = 'Integration not succesful. '
+                    message += 'Change sizes of time steps or the paramters.'
+                    raise RuntimeWarning(message)
```

Comment:
Should this be a warning or an error? My instinct would be the latter, but I'm not sure.

Reply:
Usually a warning (from my experience) ?
I can change this to an error if that fits better in the pipeline.